cargo-miri/bin.rs

```diff
@@ -646,14 +657,19 @@ fn phase_cargo_rustc(args: env::Args) {
     }
 }
 
-fn phase_cargo_runner(binary: &Path, binary_args: env::Args) {
+fn phase_cargo_runner(binary: &Path, mut binary_args: env::Args) {
     let verbose = std::env::var_os("MIRI_VERBOSE").is_some();
 
     let file = File::open(&binary)
         .unwrap_or_else(|_| show_error(format!("file {:?} not found or `cargo-miri` invoked incorrectly; please only invoke this binary through `cargo miri`", binary)));
     let file = BufReader::new(file);
-    let info: CrateRunInfo = serde_json::from_reader(file)
-        .unwrap_or_else(|_| show_error(format!("file {:?} contains outdated or invalid JSON; try `cargo clean`", binary)));
+    let info: CrateRunInfo = serde_json::from_reader(file).unwrap_or_else(|_| {
+        // FIXME: This is used to workaround that Cargo uses `CARGO_TARGET_*_RUNNER`
```

Comment:
> However, this seems like an ad-hoc work-around. I'd really prefer it we could make cargo-miri properly detect this situation upfront, instead of having to "change its mind" when the JSON read failed.

I left this as a FIXME in this PR. If I can't resolve #1677, I can't find a way better than this to fix #1660 at this moment.